Example/CotterIOS/ViewController.swift

```diff
@@ -151,6 +160,71 @@ class ViewController: UIViewController {
             break
         }
     }
+    
+    @IBAction func switchBiometric(_ sender: Any) {
+        func success(data:Data?) {
+            // toggle on success
+            print("Successfully updated Biometric enrollment")
+            guard let data = data else { return }
+            let decoder = JSONDecoder()
+            
+            do {
+                let resp = try decoder.decode(CotterUser.self, from: data)
+                
+                // check if biometric is enrolled
+                let lookFor = CotterConstants.MethodBiometric
+                var biometricAvailable = false
+                for method in resp.enrolled {
+                    if method == lookFor {
+                        biometricAvailable = true
+                        break
+                    }
+                }
+                
+                self.bioSwitch.setOn(biometricAvailable, animated: true)
+            } catch {
+                print(error.localizedDescription)
+            }
+        }
+        let cb = CotterCallback(
+            successfulFunc: success
+        )
+        CotterAPIService.shared.updateBiometricStatus(enrollBiometric: self.bioSwitch.isOn, cb: cb)
+    }
+}
+
+extension ViewController {
+    // only call setupBiometricToggle when the user is successfully been registered
+    // to the server
+    private func setupBiometricToggle() {
+        print("setting up biometric toggle")
+        func success(data:Data?) {
+            guard let data = data else { return }
+            let decoder = JSONDecoder()
+            do {
+                let resp = try decoder.decode(CotterUser.self, from: data)
+                
+                // check if biometric is enrolled
+                let lookFor = CotterConstants.MethodBiometric
+                var biometricAvailable = false
+                for method in resp.enrolled {
+                    if method == lookFor {
+                        biometricAvailable = true
+                        break
+                    }
+                }
+                
+                self.bioSwitch.setOn(biometricAvailable, animated: true
+                )
+            } catch {
+                print(error.localizedDescription)
+            }
+        }
+        let cb = CotterCallback(
+            successfulFunc: success
+        )
+        CotterAPIService.shared.getUser(userID: self.userID, cb: cb)
```

Comment:
Dude I think for checking whether to setup biometric toggle (Asked Putri about this), it should call the `CotterAPIService.shared.getBiometricStatus` function instead, because that function checks whether the current specific device (say device A) allows biometrics. (Each device = Each public key)

For the `CotterAPIService.shared.getUser` function, it checks whether any one of the user's devices (device A, B, C) allows biometrics. So if device A doesn't allow biometrics but device B does, it will still show "BIOMETRICS" in the `enrolled` array in the response.

Perhaps if needed in the future we can just add another endpoint that lets u toggle biometrics for all devices instead of just for the current device?

Reply:
To check if a user is enrolled in this device:

`GET /user/enrolled/:client_user_id/:method/:public_key`

`Response {
  enrolled: true/false,
  method:   <BIOMETRIC or PIN>,
}`